### PR TITLE
Small adjustements to let various tests pass in overloaded Fedora infra and related changes

### PR DESCRIPTION
**Pull request by keszybz**
When building in Fedora's koji, test-fs-util would fail:
```
--- command ---
10:18:29 SYSTEMD_LANGUAGE_FALLBACK_MAP='/builddir/build/BUILD/systemd-246-rc2/src/locale/language-fallback-map' PATH='/builddir/build/BUILD/systemd-246-rc2/x86_64-redhat-linux-gnu:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin' SYSTEMD_KBD_MODEL_MAP='/builddir/build/BUILD/systemd-246-rc2/src/locale/kbd-model-map' /builddir/build/BUILD/systemd-246-rc2/x86_64-redhat-linux-gnu/test-fs-util
--- stderr ---
/* test_chase_symlinks */
/* test_unlink_noerrno */
/* test_readlink_and_make_absolute */
/* test_var_tmp */
/* test_dot_or_dot_dot */
/* test_access_fd */
/* test_touch_file */
/* test_unlinkat_deallocate */
/* test_fsync_directory_of_file */
/* test_rename_noreplace */
/* test_path_is_encrypted */
/home encrypted: yes
/var encrypted: yes
/ encrypted: yes
/proc encrypted: no
/sys encrypted: no
/dev encrypted: yes
Assertion 'expect < 0 || ((r > 0) == (expect > 0))' failed at src/test/test-fs-util.c:863, function test_path_is_encrypted_one(). Aborting.
-------
```

It seems / is encrypted, but /dev is just a normal directory.

### CODE REVIEW

File: src/test/test-fs-util.c [hunk 1]
```diff
@@ -864,14 +864,16 @@ static void test_path_is_encrypted_one(const char *p, int expect) {
 }
 
 static void test_path_is_encrypted(void) {
-        log_info("/* %s */", __func__);
+        int booted = sd_booted();
```
Comment (poettering) [inline src/test/test-fs-util.c]: maybe add comment here? ("If this is run in build environments such as koji, /dev might be a reguar fs, let's protect against that?")

File: src/test/test-fs-util.c [hunk 2]
```diff
@@ -864,14 +864,16 @@ static void test_path_is_encrypted_one(const char *p, int expect) {
...
+        log_info("/* %s (sd_booted=%d)*/", __func__, booted);
 
         test_path_is_encrypted_one("/home", -1);
         test_path_is_encrypted_one("/var", -1);
         test_path_is_encrypted_one("/", -1);
-        test_path_is_encrypted_one("/proc", false);
-        test_path_is_encrypted_one("/sys", false);
-        test_path_is_encrypted_one("/dev", false);
+        test_path_is_encrypted_one("/proc", booted > 0 ? false : -1);
+        test_path_is_encrypted_one("/sys", booted > 0 ? false : -1);
```
Comment (poettering) [inline src/test/test-fs-util.c]: uh, if these two aren't the real things, then we are fucked... I wouldn't attempt to support that. Much of our API uses files in /sys and /proc, and they cannot be replaced by regular persistent file systems unlike /dev. I#d drop these two changes

File: src/test/test-fs-util.c [hunk 3]
```diff
@@ -864,14 +864,16 @@ static void test_path_is_encrypted_one(const char *p, int expect) {
...
+        test_path_is_encrypted_one("/dev", booted > 0 ? false : -1);
```
Comment (poettering) [inline src/test/test-fs-util.c:877]: a simpler fix might be to simply delete this whole line actually, given that /dev can be anything. Testing /sys and /proc should be enough

File: src/test/test-path.c [hunk 4]
```diff
@@ -79,13 +79,10 @@ static Service *service_for_path(Manager *m, Path *path, const char *service_nam
...
 
 static void check_states(Manager *m, Path *path, Service *service, PathState path_state, ServiceState service_state) {
-        usec_t ts;
-        usec_t timeout = 2 * USEC_PER_SEC;
-
         assert_se(m);
         assert_se(service);
 
-        ts = now(CLOCK_MONOTONIC);
+        usec_t ts = now(CLOCK_MONOTONIC);
```
Comment (poettering) [inline src/test/test-path.c]: if you change this around I'd always calculate the end point here, to separate this out. i.e. 

```c
usec_t end = now(COCK_MONOTONIC) + 30 * USER_PER_SEC;
```

and then further down just `if (n >= end) … `

File: src/test/test-fs-util.c (hunk 3 above)
Comment (keszybz) [reply src/test/test-fs-util.c:877]: I kept the test, because it's good to look at the corner cases.

File: src/test/test-fs-util.c (hunk 2 above)
Comment (keszybz) [reply src/test/test-fs-util.c]: Ack.

File: src/test/test-path.c (hunk 4 above)
Comment (keszybz) [reply src/test/test-path.c]: Yes, that's much better.

**Comment by keszybz** [comment]
Updated with the suggested changes (or most of them). Additional commit to log more about sleep modes.

File: src/shared/sleep-config.c [hunk 5]
```diff
@@ -125,10 +125,16 @@ int can_sleep_state(char **types) {
...
                 FOREACH_WORD_SEPARATOR(word, l, p, WHITESPACE, state)
-                        if (l == k && memcmp(word, *type, l) == 0)
+                        if (l == k && memcmp(word, *type, l) == 0) {
+                                log_debug("Sleep mode \"%s\" is supported by the kernel.", *type);
                                 return true;
+                        }
         }
 
+        if (DEBUG_LOGGING) {
+                char *t = strv_join(types, "/");
```
Comment (poettering) [inline src/shared/sleep-config.c]: leaks memory

File: src/shared/sleep-config.c [hunk 6]
```diff
@@ -125,10 +125,16 @@ int can_sleep_state(char **types) {
 
                 k = strlen(*type);
                 FOREACH_WORD_SEPARATOR(word, l, p, WHITESPACE, state)
```
Comment (poettering) [inline src/shared/sleep-config.c:127]: one of these days we should remove the use of FOREACH_WORD from our tree

Comment (keszybz) [reply src/shared/sleep-config.c:127]: Ack, I'll queue up a patch for that.

**Comment by keszybz** [comment]
Updated with the memleak fixed.
